lib/spack/spack/binary_distribution.py

```diff
@@ -0,0 +1,111 @@
+__author__ = "Benedikt Hegner (CERN)"
+
+import llnl.util.tty as tty
+
+import os
+import platform
+import urllib2
+from architecture import get_full_system_from_platform
+from spack.util.executable import which
+import spack.cmd
+import spack
+from spack.stage import Stage
+import spack.fetch_strategy as fs
+
+def prepare():
+    """
+    Install patchelf as pre-requisite to the required relocation of binary packages
+    """
+    patchelf_spec = spack.cmd.parse_specs("patchelf", concretize=True)[0]
+    patchelf = spack.repo.get(patchelf_spec)
+    patchelf.do_install()
+
+def build_info_file(spec):
+    """
+    Filename of the binary package meta-data file
+    """
+    return os.path.join(spec.prefix,".spack","binary_distribution")
+
+def tarball_name(spec):
+    """
+    Return the name of the tarfile according to the convention
+    <architecture>-<os>-<name>-<dag_hash>.tar.gz
+    """
+    return "%s-%s-%s-%s.tar.gz" %(get_full_system_from_platform(),spec.name,spec.version,spec.dag_hash())
+
+def build_tarball(spec, outdir, force=False):
+    """
+    Build a tarball from given spec
+    """
+    tarfile = os.path.join(outdir, tarball_name(spec))
+    if os.path.exists(tarfile):
+        if force:
+            os.remove(tarfile)
+        else:
+            tty.die("file exists, use -f to force overwrite: %s"%tarfile)
+
+    tar = which('tar', required=True)
+    dirname = os.path.dirname(spec.prefix)
+    basename = os.path.basename(spec.prefix)
+    
+    # handle meta-data
+    cp = which("cp", required = True)
+    spec_file = os.path.join(spec.prefix,".spack","spec.yaml")
```

Comment:
@tgamblin I am trying to use this pull request with the current llnl/develop but the path generated by spec.prefix does not match the actual install path. What is the correct path to use to find an installed package?